tests/includes/Sync/Test_Write_Controller.php

```
@@ -0,0 +1,1360 @@
+	public bool $recordLockOk = true;
+	/** @var string[] ordered trace of acquire/release_record_lock + apply, to assert the lock wraps apply */
+	public array $lockTrace = array();
+
+	public function lookup( string $collection, string $mutation_id ): ?array {
+		return $this->lookups[ $mutation_id ] ?? null;
+	}
+	public function reserve( string $collection, string $mutation_id, string $record_uuid, string $operation ): bool {
+		$this->reserved[] = $mutation_id;
+		return array_shift( $this->reserveResults ) ?? true;
+	}
+	public function mark_applied( string $mutation_id, int $remote_id, int $response_status ): bool {
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$response_status'. (undefined)

(UnusedFormalParameter)